ldtalent/mainakamau-auth-django-graphql-sendgrid
# Feature(Enable Email Confirmation): Users should confirm their emails before logging in

## What does this PR do?
It creates implementation to enable a user to confirm their email upon registration

## Description of the task to be completed
- Enable registered users to receive email confirmation emails
- Send an email link to users registered email
- Create an HTML template for the email confirmation email
- Validated email confirmation before login

## How can this be manually tested
- Clone repository to a local machine
- Use .exampleenv to set up the environment variables
- create migrations by running `python manage.py makemigrations`
- migrate by running `python manage.py migrate`
- start the development server by running `python manage.py runserver`
- access the graphql endpoint by navigating to `http:127.0.0.1:8000/sendgriddemo`
- Register a user by running 
```
mutation {
  registerUser(email: "johndoe@gmail.com", 
    password: "johndoe123", username: "johndoe") {
    user {
      email
    },
    message
  }
}
```
- Check email for confirmation email and click on it to activate the account

## How has this been tested 
- Unittest
